Submitted By: Joe Ciccone <jciccone@linuxfromscratch.org>
Date: 2006-08-20
Initial Package Version: 5.8.8
Upstream Status: Unknown
Origin: http://cvs.fedora.redhat.com/viewcvs/*checkout*/devel/perl/perl-5.8.1-fpic.patch
Description: Fixes a test that checks to see which paramater needs to be used
             for -fPIC and forces the objects in DynaLoader to be built
             with -fPIC.

diff -Naur perl-5.8.8.orig/Configure perl-5.8.8/Configure
--- perl-5.8.8.orig/Configure	2006-08-20 15:18:40.000000000 -0400
+++ perl-5.8.8/Configure	2006-08-20 15:21:39.000000000 -0400
@@ -7586,7 +7586,7 @@
 			;;
 		*)  case "$osname" in
 	                darwin) dflt='none' ;;
-			svr4*|esix*|solaris|nonstopux) dflt='-fPIC' ;;
+			linux*|svr4*|esix*|solaris|nonstopux) dflt='-fPIC' ;;
 			*)	dflt='-fpic' ;;
 		    esac ;;
 	    esac ;;
diff -Naur perl-5.8.8.orig/ext/DynaLoader/hints/linux.pl perl-5.8.8/ext/DynaLoader/hints/linux.pl
--- perl-5.8.8.orig/ext/DynaLoader/hints/linux.pl	2006-08-20 15:18:41.000000000 -0400
+++ perl-5.8.8/ext/DynaLoader/hints/linux.pl	2006-08-20 15:35:29.000000000 -0400
@@ -2,4 +2,7 @@
 # Some Linux releases like to hide their <nlist.h>
 $self->{CCFLAGS} = $Config{ccflags} . ' -I/usr/include/libelf'
 	if -f "/usr/include/libelf/nlist.h";
+# Some silly modules like mod_perl use DynaLoader.a in a shared
+# moduke, so add cccdlflags if we're going for a shared libperl
+$self->{CCFLAGS} = ($self->{CCFLAGS} || $Config{ccflags}) . " $Config{cccdlflags}";
 1;
